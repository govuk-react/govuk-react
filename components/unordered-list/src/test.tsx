import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import Link from '@govuk-react/link';
import ListItem from '@govuk-react/list-item';

import UnorderedList from '.';

const wrapper = (
  <UnorderedList>
    <ListItem>
      <Link href="http://example.com">Example 1</Link>
    </ListItem>
    <ListItem>
      <Link href="http://example.com">Example 2</Link>
    </ListItem>
    <ListItem>
      <Link href="http://example.com">Example 3</Link>
    </ListItem>
  </UnorderedList>
);

describe('UnorderedList', () => {
  it('renders with and without props, without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<UnorderedList>Example</UnorderedList>, div);
    ReactDOM.render(<UnorderedList listStyleType="number">Example</UnorderedList>, div);
  });

  it('matches wrapper snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });
});
