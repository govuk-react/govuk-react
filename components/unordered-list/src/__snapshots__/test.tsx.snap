// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnorderedList matches wrapper snapshot: wrapper mount 1`] = `
.c4 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c4:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
  background-color: #ffdd00;
}

.c4:link {
  color: #1d70b8;
}

.c4:visited {
  color: #4c2c92;
}

.c4:hover {
  color: #2b8cc4;
}

.c4:active {
  color: #2b8cc4;
}

.c4:focus {
  color: #0b0c0c;
}

.c3 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  margin-bottom: 5px;
}

.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  margin-top: 0;
  margin-bottom: 15px;
  padding-left: 20px;
}

.c0 .c0 {
  margin-top: 10px;
}

.c1 > .c2 {
  margin-bottom: 5px;
}

@media print {
  .c4 {
    font-family: sans-serif;
  }
}

@media print {
  .c4[href^="/"]::after,
  .c4[href^="http://"]::after,
  .c4[href^="https://"]::after {
    content: " (" attr(href) ")";
    font-size: 90%;
    word-wrap: break-word;
  }
}

@media print {
  .c3 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c3 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 20px;
  }
}

<UnorderedList>
  <src__OrderedList
    as="ul"
  >
    <ul
      className="c0 c1"
    >
      <src__ListItem>
        <li
          className="c2 c3"
        >
          <styled.a
            href="http://example.com"
            muted={false}
            noVisitedState={false}
            textColour={false}
          >
            <a
              className="c4"
              href="http://example.com"
            >
              Example 1
            </a>
          </styled.a>
        </li>
      </src__ListItem>
      <src__ListItem>
        <li
          className="c2 c3"
        >
          <styled.a
            href="http://example.com"
            muted={false}
            noVisitedState={false}
            textColour={false}
          >
            <a
              className="c4"
              href="http://example.com"
            >
              Example 2
            </a>
          </styled.a>
        </li>
      </src__ListItem>
      <src__ListItem>
        <li
          className="c2 c3"
        >
          <styled.a
            href="http://example.com"
            muted={false}
            noVisitedState={false}
            textColour={false}
          >
            <a
              className="c4"
              href="http://example.com"
            >
              Example 3
            </a>
          </styled.a>
        </li>
      </src__ListItem>
    </ul>
  </src__OrderedList>
</UnorderedList>
`;
